src/Features/Core/Portable/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.SymbolStartAnalyzer.BlockAnalyzer.cs

```diff
@@ -0,0 +1,492 @@
+                                                             value.Syntax.GetLocation(),
+                                                             _options.UnusedValueExpressionStatementSeverity,
+                                                             additionalLocations: null,
+                                                             properties);
+                    context.ReportDiagnostic(diagnostic);
+                }
+
+                private void AnalyzeDelegateCreationOrAnonymousFunction(OperationAnalysisContext operationAnalysisContext)
+                    => _hasDelegateCreationOrAnonymousFunction = true;
+
+                private void AnalyzeConversion(OperationAnalysisContext operationAnalysisContext)
+                {
+                    var conversion = (IConversionOperation)operationAnalysisContext.Operation;
+                    if (!_hasConversionFromDelegateTypeToNonDelegateType &&
+                        conversion.Operand.Type.IsDelegateType() &&
+                        !conversion.Type.IsDelegateType())
+                    {
+                        _hasConversionFromDelegateTypeToNonDelegateType = true;
+                    }
+                }
+
+                private void AnalyzeFieldOrPropertyReference(OperationAnalysisContext operationAnalysisContextContext)
+                {
+                    var fieldOrPropertyReference = operationAnalysisContextContext.Operation;
+                    if (!_delegateAssignedToFieldOrProperty &&
+                        fieldOrPropertyReference.Type.IsDelegateType() &&
+                        fieldOrPropertyReference.Parent is ISimpleAssignmentOperation simpleAssignment &&
+                        simpleAssignment.Target == fieldOrPropertyReference)
+                    {
+                        _delegateAssignedToFieldOrProperty = true;
+                    }
+                }
+
+                /// <summary>
+                /// Method invoked in <see cref="AnalyzeOperationBlockEnd(OperationBlockAnalysisContext)"/>
+                /// for each operation block to determine if we should analyze the operation block or bail out.
+                /// </summary>
+                private bool ShouldAnalyze(IOperation operationBlock, ISymbol owningSymbol)
+                {
+                    switch (operationBlock.Kind)
+                    {
+                        case OperationKind.None:
+                        case OperationKind.ParameterInitializer:
+                            // Skip blocks from attributes (which have OperationKind.None) and parameter initializers.
+                            // We don't have any unused values in such operation blocks.
+                            return false;
+                    }
+
+                    // We currently do not support points-to analysis, so we cannot accurately 
+                    // track delegate invocations for all cases.
```

Comment:
what is 'points to' analysis?

Reply:
I was referring to the standard pointer/points-to analysis ([https://en.wikipedia.org/wiki/Pointer_analysis](https://en.wikipedia.org/wiki/Pointer_analysis)), which is needed to accurately track locations of allocated objects and their aliasing, which enables us to determine if two symbols reference the same object instance at a given program point. I will improve the comment.

---
In reply to: [234803233](https://github.com/dotnet/roslyn/pull/30777#discussion_r234803233) [](ancestors = 234803233)